src/search_algo.hpp

```diff
@@ -270,24 +274,26 @@ loadDbIndexFromDisk(TGlobalHolder       & globalHolder,
         throw 88;
     }
 
-    if constexpr (!std::is_lvalue_reference_v<typename TGlobalHolder::TTransSbjSeqs>) // is view
-    {
-        // update view, now that underlying range is available
+    if constexpr (c_origSbjAlph == c_transAlph)
+        globalHolder.transSbjSeqs = globalHolder.indexFile.seqs | seqan3::views::type_reduce;       // no-op view
+    else
         globalHolder.transSbjSeqs = globalHolder.indexFile.seqs | seqan3::views::translate_join;
-    }
 
-    if constexpr (!std::is_lvalue_reference_v<typename TGlobalHolder::TRedSbjSeqs>) // is view
+    if constexpr (c_transAlph == c_redAlph)
+    {
+        globalHolder.redSbjSeqs = globalHolder.transSbjSeqs | seqan3::views::type_reduce;           // no-op view
+    }
+    else
     {
-        // update view, now that underlying range is available
-        if constexpr (TGlobalHolder::blastProgram == seqan::BlastProgram::BLASTN)
+        if constexpr (c_transAlph == AlphabetEnum::DNA5) // BLASTN mode
         {
             globalHolder.redSbjSeqs = globalHolder.transSbjSeqs
                                     | seqan3::views::dna_n_to_random;
         }
         else
         {
             globalHolder.redSbjSeqs = globalHolder.transSbjSeqs
-                                    | seqan3::views::deep{seqan3::views::convert<typename TGlobalHolder::TRedAlph>};
```

Comment:
Can't see why this particular change was made on mobile git right now. In general, i changed the order so that the if clauses do not begin with negated checks.
And I tried to make the checks based on alphabet and not `blastProgram`.

Reply:
Thanks